File: TarjetaEstilosVueDasboard/src/models/Button.ts
class Button{
    title:string|null;
    add_class:string|null;
    link:string|null;
    created_at:Date|null;
    updated_at:Date|null;
    deleted_at:Date|null;
    id:number|null;
    banners_id:number|null

    constructor(
        title:string|null,
        add_class:string|null,
        link:string|null,
        created_at:Date|null, 
        updated_at:Date|null, 
        deleted_at:Date|null, id:number|null=null,
        banners_id:number|null=null
    ){
        this.title = title;
        this.add_class = add_class;
        this.link = link;
        this.created_at = created_at;
        this.updated_at = updated_at;
        this.deleted_at = deleted_at;
        this.id = id;
        this.banners_id = banners_id;
    }
};
export default Button;

File: src/main/java/pe/tarjetaestilos/models/Button.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.tarjetaestilos.models;

/**
 *
 * @author julito
 */
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.Where;
import java.time.ZonedDateTime;
import com.fasterxml.jackson.annotation.JsonBackReference;


@Entity
@Table(name="button")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Where(clause = "deleted_at IS NULL")
public class Button {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(columnDefinition = "TEXT")
    private String title;
    
    @Column(columnDefinition = "TEXT")
    private String add_class;
    
    @Column(columnDefinition = "TEXT")
    private String link;
    
    @ManyToOne()
    @JoinColumn(name = "banners_id", nullable = false)
    @ToString.Exclude
    @JsonBackReference
    private Banners banners;
    
    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private ZonedDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at")
    private ZonedDateTime updatedAt;
    
    @Column(name = "deleted_at")
    private ZonedDateTime deletedAt;
    
}
